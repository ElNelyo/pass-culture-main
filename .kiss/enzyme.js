// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
