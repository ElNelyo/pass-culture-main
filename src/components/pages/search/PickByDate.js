import moment from 'moment'
import PropTypes from 'prop-types'
import React, { Component } from 'react'
import DatePicker from 'react-datepicker'

import { Icon } from 'pass-culture-shared'

class PickByDate extends Component {
  render() {
    const {
      dateFormat,
      filterDate,
      highlightedDates,
      id,
      maxDate,
      minDate,
      size,
      value,
    } = this.props

    return (
      <div className={`input is-${size} date-picker`}>
        <span>
          <DatePicker
            className="date"
            dateFormat={dateFormat}
            filterDate={filterDate}
            highlightDates={(highlightedDates || []).map(d => moment(d))}
            id={id}
            minDate={
              minDate === 'today' ? moment() : minDate && moment(minDate)
            }
            maxDate={
              maxDate === 'today' ? moment() : maxDate && moment(maxDate)
            }
            onChange={this.onChange}
            selected={value ? moment(value) : null}
          />
        </span>
        <span className="icon">
          <Icon alt="Horaires" className="input-icon" svg="ico-calendar" />
        </span>
      </div>
    )
  }
}

PickByDate.defaultProps = {
  dateFormat: 'DD/MM/YYYY',
}

PickByDate.propTypes = {
  dateFormat: PropTypes.string,
  filterDate: PropTypes.string.isRequired,
  highlightedDates: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  maxDate: PropTypes.string.isRequired,
  minDate: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
}

export default PickByDate
