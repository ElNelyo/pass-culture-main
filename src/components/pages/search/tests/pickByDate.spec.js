import React from 'react'
import { shallow } from 'enzyme'

import PickByDate from '../PickByDate'

describe('src | components | pages | search | PickByDate', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = {
        dateFormat: '',
        filterDate: jest.fn(),
        highlightedDates: '',
        id: '',
        maxDate: '',
        minDate: '',
        size: '',
        value: '',
      }

      // when
      const wrapper = shallow(<PickByDate {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
