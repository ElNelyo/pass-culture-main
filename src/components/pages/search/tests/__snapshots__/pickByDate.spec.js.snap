// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | search | PickByDate snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PickByDate
    dateFormat=""
    filterDate={[MockFunction]}
    highlightedDates=""
    id=""
    maxDate=""
    minDate=""
    size=""
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          <DatePicker
            allowSameDay={false}
            className="date"
            dateFormat=""
            dateFormatCalendar="MMMM YYYY"
            disabled={false}
            disabledKeyboardNavigation={false}
            dropdownMode="scroll"
            filterDate={[MockFunction]}
            highlightDates={Array []}
            id=""
            maxDate=""
            minDate=""
            monthsShown={1}
            nextMonthButtonLabel="Next month"
            onBlur={[Function]}
            onChange={[Function]}
            onClickOutside={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMonthChange={[Function]}
            onSelect={[Function]}
            onYearChange={[Function]}
            preventOpenOnFocus={false}
            previousMonthButtonLabel="Previous Month"
            readOnly={false}
            selected={null}
            shouldCloseOnSelect={true}
            showTimeSelect={false}
            timeCaption="Time"
            timeIntervals={30}
            withPortal={false}
          />
        </span>,
        <span
          className="icon"
        >
          <Icon
            alt="Horaires"
            className="input-icon"
            svg="ico-calendar"
          />
        </span>,
      ],
      "className": "input is- date-picker",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <DatePicker
            allowSameDay={false}
            className="date"
            dateFormat=""
            dateFormatCalendar="MMMM YYYY"
            disabled={false}
            disabledKeyboardNavigation={false}
            dropdownMode="scroll"
            filterDate={[MockFunction]}
            highlightDates={Array []}
            id=""
            maxDate=""
            minDate=""
            monthsShown={1}
            nextMonthButtonLabel="Next month"
            onBlur={[Function]}
            onChange={[Function]}
            onClickOutside={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMonthChange={[Function]}
            onSelect={[Function]}
            onYearChange={[Function]}
            preventOpenOnFocus={false}
            previousMonthButtonLabel="Previous Month"
            readOnly={false}
            selected={null}
            shouldCloseOnSelect={true}
            showTimeSelect={false}
            timeCaption="Time"
            timeIntervals={30}
            withPortal={false}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowSameDay": false,
            "className": "date",
            "dateFormat": "",
            "dateFormatCalendar": "MMMM YYYY",
            "disabled": false,
            "disabledKeyboardNavigation": false,
            "dropdownMode": "scroll",
            "filterDate": [MockFunction],
            "highlightDates": Array [],
            "id": "",
            "maxDate": "",
            "minDate": "",
            "monthsShown": 1,
            "nextMonthButtonLabel": "Next month",
            "onBlur": [Function],
            "onChange": [Function],
            "onClickOutside": [Function],
            "onFocus": [Function],
            "onKeyDown": [Function],
            "onMonthChange": [Function],
            "onSelect": [Function],
            "onYearChange": [Function],
            "preventOpenOnFocus": false,
            "previousMonthButtonLabel": "Previous Month",
            "readOnly": false,
            "selected": null,
            "shouldCloseOnSelect": true,
            "showTimeSelect": false,
            "timeCaption": "Time",
            "timeIntervals": 30,
            "withPortal": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Icon
            alt="Horaires"
            className="input-icon"
            svg="ico-calendar"
          />,
          "className": "icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "Horaires",
            "className": "input-icon",
            "svg": "ico-calendar",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            <DatePicker
              allowSameDay={false}
              className="date"
              dateFormat=""
              dateFormatCalendar="MMMM YYYY"
              disabled={false}
              disabledKeyboardNavigation={false}
              dropdownMode="scroll"
              filterDate={[MockFunction]}
              highlightDates={Array []}
              id=""
              maxDate=""
              minDate=""
              monthsShown={1}
              nextMonthButtonLabel="Next month"
              onBlur={[Function]}
              onChange={[Function]}
              onClickOutside={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMonthChange={[Function]}
              onSelect={[Function]}
              onYearChange={[Function]}
              preventOpenOnFocus={false}
              previousMonthButtonLabel="Previous Month"
              readOnly={false}
              selected={null}
              shouldCloseOnSelect={true}
              showTimeSelect={false}
              timeCaption="Time"
              timeIntervals={30}
              withPortal={false}
            />
          </span>,
          <span
            className="icon"
          >
            <Icon
              alt="Horaires"
              className="input-icon"
              svg="ico-calendar"
            />
          </span>,
        ],
        "className": "input is- date-picker",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <DatePicker
              allowSameDay={false}
              className="date"
              dateFormat=""
              dateFormatCalendar="MMMM YYYY"
              disabled={false}
              disabledKeyboardNavigation={false}
              dropdownMode="scroll"
              filterDate={[MockFunction]}
              highlightDates={Array []}
              id=""
              maxDate=""
              minDate=""
              monthsShown={1}
              nextMonthButtonLabel="Next month"
              onBlur={[Function]}
              onChange={[Function]}
              onClickOutside={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMonthChange={[Function]}
              onSelect={[Function]}
              onYearChange={[Function]}
              preventOpenOnFocus={false}
              previousMonthButtonLabel="Previous Month"
              readOnly={false}
              selected={null}
              shouldCloseOnSelect={true}
              showTimeSelect={false}
              timeCaption="Time"
              timeIntervals={30}
              withPortal={false}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowSameDay": false,
              "className": "date",
              "dateFormat": "",
              "dateFormatCalendar": "MMMM YYYY",
              "disabled": false,
              "disabledKeyboardNavigation": false,
              "dropdownMode": "scroll",
              "filterDate": [MockFunction],
              "highlightDates": Array [],
              "id": "",
              "maxDate": "",
              "minDate": "",
              "monthsShown": 1,
              "nextMonthButtonLabel": "Next month",
              "onBlur": [Function],
              "onChange": [Function],
              "onClickOutside": [Function],
              "onFocus": [Function],
              "onKeyDown": [Function],
              "onMonthChange": [Function],
              "onSelect": [Function],
              "onYearChange": [Function],
              "preventOpenOnFocus": false,
              "previousMonthButtonLabel": "Previous Month",
              "readOnly": false,
              "selected": null,
              "shouldCloseOnSelect": true,
              "showTimeSelect": false,
              "timeCaption": "Time",
              "timeIntervals": 30,
              "withPortal": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon
              alt="Horaires"
              className="input-icon"
              svg="ico-calendar"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "Horaires",
              "className": "input-icon",
              "svg": "ico-calendar",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
